### ISSUE TITLE
_lookupFactory deprecation warnings logged in Ember 2.12

### ISSUE THREAD

**Issue by acorncom** [mitchlloyd/ember-islands#45]
I'm seeing the following deprecation messages logged twice in a client app using the latest version of `ember-islands`:

```
DEPRECATION: Using "_lookupFactory" is deprecated. Please use container.factoryFor instead. [deprecation id: container-lookupFactory] See http://emberjs.com/deprecations/v2.x/#toc_migrating-from-_lookupfactory-to-factoryfor for more details.
```

From what I can see, it looks like the issue is in these lines:
https://github.com/mitchlloyd/ember-islands/blob/master/addon/components/ember-islands.js#L77-L83

A simple replacement of the offending code (per the deprecation guides) with the following:
```
  let layout = owner.factoryFor(layoutLookupKey);
```

then throws errors in the new `ember-glimmer` module:

```
Error: You attempted to access "isDescriptor" on a factory manager created by container#factoryFor. "isDescriptor" is not a member of a factory manager."
```

Those lines are being triggered in the `templateFor` method here:
https://github.com/emberjs/ember.js/blob/v2.12.0/packages/ember-glimmer/lib/syntax/curly-component.js#L247-L270

I'll probably be back to follow up on this further, but thought I'd post what I found so far ...

**Comment by RecoX** [comment]
@mitchlloyd Cmon man, accept the PR of this @acorncom , its painful see that error :(

**Comment by acorncom** [comment]
@RecoX sadly, there isn't a PR on this yet, it gets a bit tricky in there with Glimmer ...

**Comment by RecoX** [comment]
@acorncom  Sorry for my ignorance, but can I ask why is it tricky ? I mean, in your comment everything was like almost done, changing the deprecated method with the new one as you said.

**Comment by acorncom** [comment]
@RecoX if it had actually worked, I would have sent in a PR 😀 Unfortunately, I ran out of time to debug things further at the time, but I was staring down a deeper hole than a simple deprecation fix. If you apply these fixes to your setup, you'll see a number of other errors that are worse 😀

**Comment by mitchlloyd** [comment]
This issue is becoming dire now that `_lookupFactory` has been removed from Ember 2.15. @RecoX could you investigate?

The deprecations guide has [some useful-looking information](https://www.emberjs.com/deprecations/v2.x/#toc_id-container-lookupfactory). Perhaps we just need to use this new `.class` API for something after the switch to `factoryFor` (first time I've seen that). Also that guide mentions that an upgrade to `ember-getowner-polyfill` should make the new API work with older versions of Ember, which we may need to keep all the tests passing.

**Comment by mitchlloyd** [comment]
Published in 1.3.0
